src/main/java/com/github/junrar/Archive.java

```diff
@@ -425,30 +417,22 @@ private void readHeaders(final long fileLength) throws IOException, RarException
                                 - BlockHeader.BaseBlockSize
                                 - BlockHeader.blockHeaderSize;
                             final byte[] protectHeaderBuffer = safelyAllocate(toRead, MAX_HEADER_SIZE);
-                            this.channel.readFully(protectHeaderBuffer, toRead);
-                            final ProtectHeader ph = new ProtectHeader(blockHead,
-                                protectHeaderBuffer);
-                            newpos = ph.getPositionInFile() + ph.getHeaderSize()
-                                + ph.getDataSize();
-                            if (processedPositions.contains(newpos)) {
```

Comment:
I see a few places where this is removed. Do you know what this was doing, and why we don't need it anymore?

Reply:
oh,sorry.
I didn't see “processedPositions” in the previous code。I thought this was used to unpack header-encrypted archive.
I just read the comment and find it's used to track the corruption.
I will restore it, don't worry。